Resource Manager Model

EvenCode opened the pull request:
Problem with Composition. Check ML.

Compro-Prasad committed:
Resource view GUI

Compro-Prasad committed:
Added basic functions in resources component

Compro-Prasad committed:
Fix errors related to left out WBS args in resources component

Compro-Prasad committed:
Define all the functions for resource manager component

Compro-Prasad committed:
Fix all errors in resource manager component

Commit:
Resource Manager Model

Compro-Prasad committed:
Basic layout of Resource view created

Commit:
RM class relationships

Commit:
add group and resource modified

Commit:
iterator problem resolved

Commit:
id generation for resource

Commit:
id generation for resource group

Commit:
res-manager add_group now returns ref to new group

Commit:
.

Compro-Prasad committed:
Merge remote-tracking branch 'kapil/rm-model' into resource

Compro-Prasad committed:
Resource manager controller

Compro-Prasad commented:
Add `add_group` in the `ResourceManager` class.

Commit:
Resource Manager can now add resources

Compro-Prasad commented on src/resource-manager.h:
```diff
@@ -22,7 +22,17 @@ class ResourceManager
+    {
+      if (it->id() == id)
+      {
+        return (it->ResourceGroup::add_resource());
```
`ResourceGroup::` isn't necessary

Compro-Prasad commented on src/resource-manager.h:
```diff
@@ -22,7 +22,17 @@ class ResourceManager
 public:
   const ResourceGroup & add_group (void);
-
+  const Resource & add_resource (const std::string & id)
```
make it `group_id` instead of `id`

Commit:
.

Compro-Prasad commented on src/resource-manager.cxx:
```diff
@@ -11,6 +11,19 @@
+ResourceGroup
+ResourceManager::get_group (const std::string & id) const
+{
+  for (auto it = m_resource_groups.begin(); it != m_resource_groups.end(); it++)
```
There is also `for (auto &i : m_resource_groups)`

Compro-Prasad commented on src/resource-manager.cxx:
```diff
@@ -11,6 +11,19 @@
+  {
+    if (it->id() == id)
+    {
+      return *it;
```
Return pointer to the `ResourceGroup`

Commit:
Resource Manager - add_resource()

Compro-Prasad committed:
Merge remote-tracking branch 'kapil/rm-model' into resource

Compro-Prasad committed:
Added support for adding a resource through view

Commit:
methods for changing resource and resource_group attributes in resource_group and resource_manager

Commit:
validation and generation of group and resource id to be unique.

Compro-Prasad committed:
Merge branch 'kapil/rm-model' into resource and fix the controller

Compro-Prasad committed:
Full abstraction between view, model and controller completed

- add_{resource, group} now return index and not actual object as reference
- Model is expected to be accessed using index only though there are functions
  that return actual Resource and ResourceGroup objects which are more
  efficient and easier to use

Commit:
random number generator improved
